File: sample/app/src/main/java/com/paytabs/sample/paytabstestsample/MainActivity.java
package com.paytabs.sample.paytabstestsample;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import com.paytabs.paytabs_sdk.payment.ui.activities.PayTabActivity;
import com.paytabs.paytabs_sdk.utils.PaymentParams;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.pay).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                goPayment();
            }
        });
    }

    private void goPayment() {
      Intent in = new Intent(getApplicationContext(), PayTabActivity.class);
      in.putExtra(PaymentParams.MERCHANT_EMAIL, "");
      in.putExtra(PaymentParams.SECRET_KEY,"");//Add your Secret Key Here
      in.putExtra(PaymentParams.LANGUAGE,PaymentParams.ENGLISH);
      in.putExtra(PaymentParams.TRANSACTION_TITLE, "Test Paytabs android library");
      in.putExtra(PaymentParams.AMOUNT, 5.0);

      in.putExtra(PaymentParams.CURRENCY_CODE, "BHD");
      in.putExtra(PaymentParams.CUSTOMER_PHONE_NUMBER, "009733");
      in.putExtra(PaymentParams.CUSTOMER_EMAIL, "customer-email@example.com");
      in.putExtra(PaymentParams.ORDER_ID, "123456");
      in.putExtra(PaymentParams.PRODUCT_NAME, "Product 1, Product 2");

      //Billing Address
      in.putExtra(PaymentParams.ADDRESS_BILLING, "Flat 1,Building 123, Road 2345");
      in.putExtra(PaymentParams.CITY_BILLING, "Manama");
      in.putExtra(PaymentParams.STATE_BILLING, "Manama");
      in.putExtra(PaymentParams.COUNTRY_BILLING, "BHR");
      in.putExtra(PaymentParams.POSTAL_CODE_BILLING, "00973"); //Put Country Phone code if Postal code not available '00973'

      //Shipping Address
      in.putExtra(PaymentParams.ADDRESS_SHIPPING, "Flat 1,Building 123, Road 2345");
      in.putExtra(PaymentParams.CITY_SHIPPING, "Manama");
      in.putExtra(PaymentParams.STATE_SHIPPING, "Manama");
      in.putExtra(PaymentParams.COUNTRY_SHIPPING, "BHR");
      in.putExtra(PaymentParams.POSTAL_CODE_SHIPPING, "00973"); //Put Country Phone code if Postal code not available '00973'


       // force shipping data validation
        in.putExtra(PaymentParams.FORCE_SHIPPING_VALIDATION, true);

      //Region
      in.putExtra(PaymentParams.REGION_ENDPOINT, PaymentParams.UAE_REGION);

      // for Samsung pay integration
     /* in.putExtra(PaymentParams.SAMSUNG_PAY_JSON, samPaytoken);
      in.putExtra(PaymentParams.CUSTOMER_NAME, customerName);*/

      //Payment Page Style
      in.putExtra(PaymentParams.PAY_BUTTON_COLOR, "#2474bc");

      //Tokenization
      in.putExtra(PaymentParams.IS_TOKENIZATION, true);
      startActivityForResult(in, PaymentParams.PAYMENT_REQUEST_CODE);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (resultCode == RESULT_OK && requestCode == PaymentParams.PAYMENT_REQUEST_CODE) {
            Log.e("Tag", data.getStringExtra(PaymentParams.RESPONSE_CODE));
            Log.e("Tag", data.getStringExtra(PaymentParams.TRANSACTION_ID));
            Log.e("Tag STATEMENT_REFERENCE", data.getStringExtra(PaymentParams.STATEMENT_REFERENCE)+"");
            Log.e("Tag TRACE_CODE", data.getStringExtra(PaymentParams.TRACE_CODE)+"");
            if (data.hasExtra(PaymentParams.TOKEN) && !data.getStringExtra(PaymentParams.TOKEN).isEmpty()) {
                Log.e("Tag", data.getStringExtra(PaymentParams.TOKEN));
                Log.e("Tag", data.getStringExtra(PaymentParams.CUSTOMER_EMAIL));
                Log.e("Tag", data.getStringExtra(PaymentParams.CUSTOMER_PASSWORD));
            }
        }
    }
}

File: sample-kotlin/app/src/main/java/com/paytabs/sample/kotlin/MainActivity.kt
package com.paytabs.sample.kotlin

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import com.paytabs.paytabs_sdk.payment.ui.activities.PayTabActivity
import com.paytabs.paytabs_sdk.utils.PaymentParams
import androidx.core.app.ComponentActivity
import androidx.core.app.ComponentActivity.ExtraData
import androidx.core.content.ContextCompat.getSystemService
import android.icu.lang.UCharacter.GraphemeClusterBreak.T
import android.util.Log



class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<Button>(R.id.pay).setOnClickListener(View.OnClickListener { goPayment() })

    }

    fun goPayment() {
        val intent = Intent(applicationContext, PayTabActivity::class.java)
        intent.putExtra(PaymentParams.MERCHANT_EMAIL, "")
        intent.putExtra(
            PaymentParams.SECRET_KEY,
            ""
        )//Add your Secret Key Here
        intent.putExtra(PaymentParams.LANGUAGE, PaymentParams.ENGLISH)
        intent.putExtra(PaymentParams.TRANSACTION_TITLE, "Test Paytabs android library")
        intent.putExtra(PaymentParams.AMOUNT, 5.0)

        intent.putExtra(PaymentParams.CURRENCY_CODE, "BHD")
        intent.putExtra(PaymentParams.CUSTOMER_PHONE_NUMBER, "009733")
        intent.putExtra(PaymentParams.CUSTOMER_EMAIL, "customer-email@example.com")
        intent.putExtra(PaymentParams.ORDER_ID, "123456")
        intent.putExtra(PaymentParams.PRODUCT_NAME, "Product 1, Product 2")

        //Billing Address
        intent.putExtra(PaymentParams.ADDRESS_BILLING, "Flat 1,Building 123, Road 2345")
        intent.putExtra(PaymentParams.CITY_BILLING, "Manama")
        intent.putExtra(PaymentParams.STATE_BILLING, "Manama")
        intent.putExtra(PaymentParams.COUNTRY_BILLING, "BHR")
        intent.putExtra(
            PaymentParams.POSTAL_CODE_BILLING,
            "00973"
        ) //Put Country Phone code if Postal code not available '00973'

        //Shipping Address
        intent.putExtra(PaymentParams.ADDRESS_SHIPPING, "Flat 1,Building 123, Road 2345")
        intent.putExtra(PaymentParams.CITY_SHIPPING, "Manama")
        intent.putExtra(PaymentParams.STATE_SHIPPING, "Manama")
        intent.putExtra(PaymentParams.COUNTRY_SHIPPING, "BHR")
        intent.putExtra(
            PaymentParams.POSTAL_CODE_SHIPPING,
            "00973"
        ) //Put Country Phone code if Postal code not available '00973'

        // force shipping data validation
        intent.putExtra(PaymentParams.FORCE_SHIPPING_VALIDATION, true)

        //Payment Page Style
        intent.putExtra(PaymentParams.PAY_BUTTON_COLOR, "#2474bc")

        //Tokenization
        intent.putExtra(PaymentParams.IS_TOKENIZATION, true)
        startActivityForResult(intent, PaymentParams.PAYMENT_REQUEST_CODE)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (resultCode == Activity.RESULT_OK && requestCode == PaymentParams.PAYMENT_REQUEST_CODE) {
            Log.e("Tag", data!!.getStringExtra(PaymentParams.RESPONSE_CODE))
            Log.e("Tag", data.getStringExtra(PaymentParams.TRANSACTION_ID))
            if (data.hasExtra(PaymentParams.TOKEN) && !data.getStringExtra(PaymentParams.TOKEN)!!.isEmpty()) {
                Log.e("Tag", data.getStringExtra(PaymentParams.TOKEN))
                Log.e("Tag", data.getStringExtra(PaymentParams.CUSTOMER_EMAIL))
                Log.e("Tag", data.getStringExtra(PaymentParams.CUSTOMER_PASSWORD))
            }
        }
    }
}
